Examples/PcapPlusPlus-benchmark/benchmark-google.cpp

```diff
@@ -0,0 +1,236 @@
+	if (!reader.open())
+	{
+		state.SkipWithError("Cannot open pcap file for reading");
+		return;
+	}
+
+	pcpp::RawPacket rawPacket;
+	for (auto _ : state)
+	{
+		if (!reader.getNextPacket(rawPacket))
+		{
+			// If the rawPacket is empty there should be an error
+			if (totalBytes == 0)
+			{
+				state.SkipWithError("Cannot read packet");
+				return;
+			}
+
+			// Rewind the file if it reached the end
+			state.PauseTiming();
+			reader.close();
+			reader.open();
+			state.ResumeTiming();
+			continue;
+		}
+
+		// Parse packet
+		pcpp::Packet parsedPacket(&rawPacket);
+
+		// Use parsedPacket to prevent compiler optimizations
+		assert(parsedPacket.getFirstLayer());
+
+		// Count total bytes and packets
+		++totalPackets;
+		totalBytes += rawPacket.getRawDataLen();
+	}
+
+	// Set statistics to the benchmark state
+	state.SetBytesProcessed(totalBytes);
+	state.SetItemsProcessed(totalPackets);
+}
+BENCHMARK(BM_PacketParsing);
+
+static void BM_PacketCrafting(benchmark::State& state)
+{
+	size_t totalBytes = 0;
+	size_t totalPackets = 0;
+
+	for (auto _ : state)
+	{
+		uint8_t randNum = static_cast<uint8_t>(rand() % 256);
+
+		pcpp::Packet packet;
+
+		// Generate random MAC addresses
+		pcpp::MacAddress srcMac(randNum, randNum, randNum, randNum, randNum, randNum);
+		pcpp::MacAddress dstMac(randNum, randNum, randNum, randNum, randNum, randNum);
+		packet.addLayer(new pcpp::EthLayer(srcMac, dstMac), true);
+
+		// Randomly choose between IPv4 and IPv6
+		if (randNum % 2)
+		{
+			packet.addLayer(new pcpp::IPv4Layer(randNum, randNum), true);
+		}
+		else
+		{
+			std::array<uint8_t, 16> srcIP = { randNum, randNum, randNum, randNum, randNum, randNum, randNum, randNum,
+				                              randNum, randNum, randNum, randNum, randNum, randNum, randNum, randNum };
+			std::array<uint8_t, 16> dstIP = { randNum, randNum, randNum, randNum, randNum, randNum, randNum, randNum,
+				                              randNum, randNum, randNum, randNum, randNum, randNum, randNum, randNum };
+
+			packet.addLayer(new pcpp::IPv6Layer(srcIP, dstIP), true);
+		}
+
+		// Randomly choose between TCP and UDP
+		if (randNum % 2)
+		{
+			packet.addLayer(new pcpp::TcpLayer(randNum % 65536, randNum % 65536), true);
+		}
+		else
+		{
+			packet.addLayer(new pcpp::UdpLayer(randNum % 65536, randNum % 65536), true);
+		}
+
+		// Calculate all fields to update the packet
+		packet.computeCalculateFields();
+
+		// Count total bytes and packets
+		++totalPackets;
+		totalBytes += packet.getRawPacket()->getRawDataLen();
+	}
+
+	// Set statistics to the benchmark state
+	state.SetBytesProcessed(totalBytes);
+	state.SetItemsProcessed(totalPackets);
+}
+BENCHMARK(BM_PacketCrafting);
+
+int main(int argc, char** argv)
+{
+	// Initialize the benchmark
+	benchmark::Initialize(&argc, argv);
+
+	// Parse command line arguments to find the pcap file name
+	for (int idx = 1; idx < argc; ++idx)
+	{
+		if (strcmp(argv[idx], "--pcap-file") == 0)
```

Comment:
Since `--pcap-file` is a mandatory parameter (and the only param for now), maybe we don't need the param itself and can run the benchmark like this:
```
.\BenchmarkExampleGoogle.exe D:\my_packets.pcap
```

Reply:
@seladb The problem is arguments are also used by google benchmark. It supports some flags as default, for example `--benchmark_filter` or `--benchmark_out_format` etc.

Reply:
Oh ok, so we can keep it 👍